crates/broker/src/order_picker.rs

```diff
@@ -332,35 +359,47 @@ where
         }
 
         let one_mill = U256::from(1_000_000);
+        let total_cycles = U256::from(proof_res.stats.total_cycles);
+
+        let (min_price, max_price) = if slashed_unexpired {
+            let price = order.request.offer.lockStake / U256::from(FRACTION_STAKE_REWARD);
+            (price, price)
+        } else {
+            (U256::from(order.request.offer.minPrice), U256::from(order.request.offer.maxPrice))
+        };
 
-        let mcycle_price_min = (U256::from(order.request.offer.minPrice)
-            / U256::from(proof_res.stats.total_cycles))
-            * one_mill;
-        let mcycle_price_max = (U256::from(order.request.offer.maxPrice)
-            / U256::from(proof_res.stats.total_cycles))
-            * one_mill;
+        let mcycle_price_min = (min_price / total_cycles) * one_mill;
+        let mcycle_price_max = (max_price / total_cycles) * one_mill;
 
         tracing::info!(
             "Order price: min: {} max: {} - cycles: {} - mcycle price: {} - {} - stake: {}",
-            format_ether(U256::from(order.request.offer.minPrice)),
-            format_ether(U256::from(order.request.offer.maxPrice)),
+            format_ether(U256::from(min_price)),
+            format_ether(U256::from(max_price)),
             proof_res.stats.total_cycles,
             format_ether(mcycle_price_min),
             format_ether(mcycle_price_max),
             order.request.offer.lockStake,
         );
 
+        // If configured to do so then take the order regardless of it is profitable
+        let ignore_profitability = if slashed_unexpired && fill_slashed_orders_altruistically {
+            tracing::info!("Ignoring profitability check for order {order_id:x}");
+            true
+        } else {
+            false
+        };
+
         // Skip the order if it will never be worth it
-        if mcycle_price_max < config_min_mcycle_price {
+        if mcycle_price_max < config_min_mcycle_price && !ignore_profitability {
             tracing::warn!("Removing under priced order {order_id:x}");
             self.db.skip_order(order_id).await.context("Failed to delete order")?;
             return Ok(false);
         }
 
-        if mcycle_price_min >= config_min_mcycle_price {
+        if mcycle_price_min >= config_min_mcycle_price || ignore_profitability {
```

Comment:
One issue is that we schedule the lock expired orders to be locked via the below `self.db.set_order_lock`. But its not actually possible to lock these orders anyone- they have to just be directly fulfilled without locking

Reply:
Added a new return type from order pricing that allows specifying what to do with the order (Lock, Skip, ProveImmediate). The code calling the picker can then update the order state in the db accordingly